### PR TITLE
Refactor openshift_version setting

### PR DESCRIPTION
**Pull request by michaelgugino**
Currently, we do lots of manipulation of user-provided
version-related variables.  Often times this manipulation
is undesired and incorrect.

This commit refactors openshift_version to not manipulate
so many things.  We will trust that the user has put
correct variables in their inventory to get the desired
outcome.

Implements: https://trello.com/c/sL9bMo3g/644-refactor-version-handling-to-new-format

### CODE REVIEW

**Comment by michaelgugino** [comment]
@smarterclayton added default for openshift_release here, PTAL.

File: roles/openshift_repos/tasks/rhel_repos.yml [hunk 1]
```diff
@@ -26,7 +26,7 @@
   command: subscription-manager repos \
                --enable="rhel-7-server-rpms" \
                --enable="rhel-7-server-extras-rpms" \
-               --enable="rhel-7-server-ose-{{ (openshift_release | default('')).split('.')[0:2] | join('.') }}-rpms" \
+               --enable="rhel-7-server-ose-{{ (openshift_version | default('')).split('.')[0:2] | join('.') }}-rpms" \
```
Comment (sdodson) [inline roles/openshift_repos/tasks/rhel_repos.yml]: Isn't openshift_version always set now? get rid of the default here.

File: roles/openshift_version/tasks/first_master.yml [hunk 2]
```diff
@@ -11,17 +11,20 @@
   - openshift_version is not defined or openshift_version == ""
   - openshift_protect_installed_version | bool
 
-- include_tasks: "{{ l_first_master_version_task_file }}"
+- name: Set openshift_version to openshift_release if undefined
+  set_fact:
+    openshift_version: "{{ openshift_release | default('3.10') }}"
```
[the remark this answers is not in the document]
Comment (sdodson) [reply roles/openshift_version/tasks/first_master.yml]: I thought our intent was to require it.

File: playbooks/openshift-node/private/image_prep.yml [hunk 3]
```diff
@@ -2,13 +2,8 @@
...
-    skip_version: True
     skip_sanity_checks: True
     skip_validate_hostnames: True
-
-- name: determine version
-  import_playbook: ../../init/version.yml
-  vars:
     l_openshift_version_determine_hosts: "oo_nodes_to_config"
     l_openshift_version_set_hosts: "all:!all"
     l_openshift_version_check_hosts: "all:!all"
```
[the remark this answers is not in the document]
Comment (mtnbikenc) [reply playbooks/openshift-node/private/image_prep.yml]: I see now that they are removed in prerequisites.yml.

File: roles/openshift_version/tasks/first_master.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (mtnbikenc) [reply roles/openshift_version/tasks/first_master.yml]: Add this to the list of things we have to update when we fork a release.

[the remark this answers is not in the document]
Comment (michaelgugino) [reply roles/openshift_version/tasks/first_master.yml]: @mtnbikenc I was thinking of adding some kind of standard comment for this type of thing, what do you think?

Something like
'# meta: update-on-branch'

Or something super-cool and smart looking.  Basically we can just grep for that string and not mind so much.

[the remark this answers is not in the document]
Comment (mtnbikenc) [reply roles/openshift_version/tasks/first_master.yml]: That would be really cool.  Nice to have a simple doc that outlines those 'update-on-branch' steps.We have the same type of scenario when we switch Ansible versions.

[the remark this answers is not in the document]
Comment (smarterclayton) [reply roles/openshift_version/tasks/first_master.yml]: Thank you.  This makes it much clearer what the installer is trying to do.  Our master branch has only partially supported multiple releases, starting in 3.10, 3.11, and 3.12 with all the upcoming changes I think it's very likely that we are going to be unable to install old versions as we converge on the new structure of the codebase.  Having this defaulting is the right thing for a large chunk of our users (check out a branch, run install, expect it to work).  We are already paying the cost to have RPMs available for multiple versions, have multiple branches and have multiple branch jobs.

**Comment by smarterclayton** [comment]
I'm going to add a GCP 3 master, 1 infra, 1 node deployment PR job that can test on merge.

**Pull request merged**